src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs

```diff
@@ -36,6 +40,60 @@ public class NUnitCompositeWorkItemSkipChildrenIntegration
         /// <returns>Calltarget state value</returns>
         public static CallTargetState OnMethodBegin<TTarget, TSuite, TResultState>(TTarget instance, TSuite testSuite, TResultState resultState, string message)
         {
+            if (testSuite is not null)
+            {
+                string typeName = testSuite.GetType().Name;
+
+                if (typeName == "CompositeWorkItem")
+                {
+                    // In case we have a CompositeWorkItem we check if there is a OneTimeSetUp failure
+                    var compositeWorkItem = testSuite.DuckCast<ICompositeWorkItem>();
+
+                    if (compositeWorkItem.Result?.ResultState?.Status == TestStatus.Failed && compositeWorkItem.Result.ResultState.Site == FailureSite.SetUp)
+                    {
+                        foreach (var item in compositeWorkItem.Children)
+                        {
+                            if (item.GetType().Name == "CompositeWorkItem")
+                            {
+                                var compositeWorkItem2 = item.DuckCast<ICompositeWorkItem>();
+                                foreach (var item2 in compositeWorkItem2.Children)
+                                {
+                                    var testResult = item2.DuckCast<IWorkItem>().Result;
+                                    Scope scope = NUnitIntegration.CreateScope(testResult.Test, typeof(TTarget));
+                                    scope.Span.Error = true;
+                                    scope.Span.SetTag(Tags.ErrorMsg, compositeWorkItem.Result.Message);
```

Comment:
Just to double check, did you intend for these tags to come from `compositeWorkItem` or did you mean to use `compositeWorkItem2`?

Reply:
yes, the first `compositeWorkItem` contains the actual exception from the `OneTimeSetup` method, so what I do is to loop over the children and tag with the same exception from the setup method.